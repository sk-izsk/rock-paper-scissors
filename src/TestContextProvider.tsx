import React from 'react';
import { ThemeProvider } from 'react-jss';
import { theme } from './theme';

const TestContextProvider: React.FC = ({ children }) => {
  return <ThemeProvider theme={theme}>{children}</ThemeProvider>;
};

export { TestContextProvider };
