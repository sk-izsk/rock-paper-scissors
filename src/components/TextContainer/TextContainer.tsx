import React from 'react';
import { createUseStyles } from 'react-jss';
import { Body1, H6 } from 'ui-neumorphism';
import { CustomTheme } from '../../theme';

export interface TextContainerProps {
  header?: string;
}

const useStyles = createUseStyles((theme: CustomTheme) => ({
  header: {
    fontWeight: 'bold',
    marginBottom: theme.spacing(1),
  },
  body: {
    textAlign: 'left',
  },
}));

const TextContainer: React.FC<TextContainerProps> = ({ header, children }) => {
  const classes = useStyles();
  return (
    <div>
      <H6 className={classes.header}>{header}</H6>
      <Body1 className={classes.body}>{children}</Body1>
    </div>
  );
};

export { TextContainer };
