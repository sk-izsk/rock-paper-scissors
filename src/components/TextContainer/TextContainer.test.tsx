import { render } from '@testing-library/react';
import React from 'react';
import { TestContextProvider } from '../../TestContextProvider';
import { TextContainer } from './TextContainer';

test('TextContainer renders correctly', () => {
  const { asFragment } = render(
    <TestContextProvider>
      <TextContainer header='mock header'>Mock description</TextContainer>
    </TestContextProvider>,
  );
  expect(asFragment()).toMatchSnapshot();
});
