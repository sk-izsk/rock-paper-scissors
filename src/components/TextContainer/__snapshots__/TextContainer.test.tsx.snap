// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextContainer renders correctly 1`] = `
<DocumentFragment>
  <div>
    <h6
      class="_3yPwv _2wZct _11JjB _1uvOH header-0-2-1"
    >
      mock header
    </h6>
    <p
      class="_3yPwv _NwSHs _11JjB _1uvOH body-0-2-2"
    >
      Mock description
    </p>
  </div>
</DocumentFragment>
`;
